### PR TITLE
Add flag to exclude course from catalogue

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

We want to allow excluding a course from the search page.

## Proposal

I propose to add a flag "is_listed" on the course model and index this field in Elasticsearch. While searching, I filter out from results the courses that have this flag set.

The flag can be set/unset from the "Course settings..." item in the "page" administration menu visible by staff users on any course detail page.

:information_source: Another option was to skip indexing for courses that have the "is_listed" flag unset. I abandonned this option because it was fragile. Indeed, a course that was previously listed (and thus indexed) had to be deleted from the index when it was switched to "not listed", leading to complications...

### CODE REVIEW

**Comment by sampaccoud** [comment]
@jbpenrath I fixed the DjangoCMS 3.8 bugs and the autocomplete search. So it should be ready for further review!

File: src/richie/apps/courses/models/course.py [hunk 1]
```diff
@@ -133,6 +133,11 @@ class Course(BasePageExtension):
         blank=True,
         null=True,
     )
+    is_listed = models.BooleanField(
+        default=True,
+        verbose_name=_("is listed"),
```
Comment (jbpenrath) [inline src/richie/apps/courses/models/course.py:138]: Just a nitpick but there is no related translation.

**Review by jbpenrath — APPROVED** [review]
Works great ! 🎉

File: src/richie/apps/courses/models/course.py (hunk 1 above)
Comment (sampaccoud) [reply src/richie/apps/courses/models/course.py:138]: True! The string will only be available in Crowdin after I merge this PR :man_shrugging:

**Pull request merged**